pkg/internal/sdk/template/brc29/brc29_address.go

```diff
@@ -0,0 +1,52 @@
+package brc29
+
+import (
+	"fmt"
+
+	"github.com/4chain-ag/go-wallet-toolbox/pkg/internal/sdk"
+	"github.com/bsv-blockchain/go-sdk/script"
+)
+
+// Address generates a blockchain address according to BRC29 specification.
+//
+// The sender key can be a private key hex string or a key deriver or ec.PrivateKey.
+// The recipient key can be a public key hex or a key deriver or ec.PublicKey.
+//
+// Additional options allow setting the address network to mainnet or testnet.
+func Address[SenderKey CounterpartyPrivateKey, RecipientKey CounterpartyPublicKey](sender SenderKey, keyID KeyID, recipient RecipientKey, opts ...func(*lockOptions)) (*script.Address, error) {
+	options := &lockOptions{
+		mainNet: true,
+	}
+
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if err := keyID.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid key ID: %w", err)
+	}
+
+	senderKeyDeriver, err := toKeyDeriver(sender)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create sender key deriver: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to create sender key deriver (input type %T): %w", sender, err)
```
if this can fail due to invalid type would be helpful to log the type WDYT ?